services/genpassw/genpassw.go

```diff
@@ -0,0 +1,77 @@
+package genpassw
+
+import (
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	alphab string = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	nums          = []int{1, 2, 3, 4, 5, 6, 7, 8, 9}
+	spChar string = "&'(-_à)=~$\"*%/!:;,<>"
+)
+
+//PwType interface that holds the methods available to generate a password
+type PwType interface {
+	GenerateChars() string
+	GenerateInts() string
+	GenerateSpchars() string
+	GenerateComplex() string
+}
+
+//Password holds the length of the password
+type Password struct {
+	length int
+}
+
+//New returns Password struct with a specified length
+func New(length int) *Password {
+	return &Password{length}
+}
+
+//GenerateChars generate password based on characters
+func (p *Password) GenerateChars() string {
+	var genStr []string
+	rand.Seed(time.Now().UnixNano())
+	for i := 0; i < p.length; i++ {
+		RI := rand.Intn(len(alphab))
+		genStr = append(genStr, string(alphab[RI]))
+	}
+	return strings.Join(genStr, "")
+}
+
+//GenerateInts generate password base on integers
+func (p *Password) GenerateInts() string {
+	var genInt []string
+	rand.Seed(time.Now().UnixNano())
```

Comment:
I'm not sure but renewing the seed each time would link the time to the seed and give to an attacker the analysis elements for cracking the randomness. Try a quick search on that topic and tell me what you found about it.